tests/src/test/scala/cats/tests/FoldableSuite.scala

```diff
@@ -554,6 +550,20 @@ class FoldableSuiteAdditional extends CatsSuite with ScalaVersionSpecificFoldabl
     checkSlidingNStackSafety(xs => NonEmptyVector.fromVectorUnsafe(xs.toVector))
   )
 
+  test("Foldable[List] doesn't break substitution") {
+    val result = List.range(0, 10).foldM(List.empty[Int])((accum, elt) => Eval.always(elt :: accum))
+
+    assert(result.value == result.value)
```

Comment:
I see that it wasn't a goal of this PR, but could we use `assertEquals` instead of `assert`? Then we would have pretty-printed diff on errors for free.

Reply:
Oh.. This test looks changed because I moved many of its neighbors into  `FoldableSuiteAdditionalStreamSpecific` below, but this one remained in the place since it is not related to `Stream`. So I just didn't touch it actually.

Regarding your question: this particular assertion looks quite weird to me because it compares `result.value` with itself. Therefore, to me it looks like as it always produces `true` and changing it to `assertEquals` wouldn't make any difference. Perhaps, I'm just not getting what it was supposed to test. Could you suggest please?

Reply:
I was talking about not only the current assertion, rather about every `assert` as a whole.

Reply:
I agree with you in general – using `assertEquals(a, b)` instead of `assert(a == b)` or even `assert(a === b)` should be a better choice in most cases. I really wonder why Cats tests prefer using the latter assertion style.

If you believe it would be a good idea, I am up to replace `assert` with `assertEquals` in those test cases which are directly affected by the PR's primary goal – I'd like to have `assertEquals` there too.

But I would hesitate to do it for all the Cats tests in general. It would blow up the scope of this PR and would mimic its primary changes. I think it is worth a separate PR – wdyt?

Reply:
Probably you're right, it's better to do it separately. By and large, I can tackle that issue.